src/Resources/config/core.php

```diff
@@ -0,0 +1,235 @@
+            ->public()
+
+        ->alias(NoopLabelTranslatorStrategy::class, 'sonata.admin.label.strategy.noop')
+
+        ->set('sonata.admin.label.strategy.underscore', UnderscoreLabelTranslatorStrategy::class)
+            ->public()
+
+        ->alias(UnderscoreLabelTranslatorStrategy::class, 'sonata.admin.label.strategy.underscore')
+
+        ->set('sonata.admin.label.strategy.form_component', FormLabelTranslatorStrategy::class)
+            ->public()
+
+        ->alias(FormLabelTranslatorStrategy::class, 'sonata.admin.label.strategy.form_component')
+
+        // NEXT_MAJOR: Remove this service.
+        ->set('sonata.admin.translator.extractor.jms_translator_bundle', DeprecatedAdminExtractor::class)
+            ->public()
+            ->tag('jms_translation.extractor', [
+                'alias' => 'sonata_admin',
+            ])
+            ->deprecate(sprintf(
+                'The service "%%service_id%%" is deprecated in favor of the "%s" service since version 3.72 and will be removed in 4.0.',
+                Sonata\AdminBundle\Translator\Extractor\AdminExtractor::class
+            ))
+            ->args([
+                new ReferenceConfigurator('sonata.admin.pool'),
+                (new ReferenceConfigurator('logger'))->nullOnInvalid(),
+            ])
+            ->call('setBreadcrumbsBuilder', [
+                new ReferenceConfigurator('sonata.admin.breadcrumbs_builder'),
+            ])
+
+        ->set(AdminExtractor::class)
+            ->tag('translation.extractor', [
+                'alias' => 'sonata_admin',
+            ])
+            ->args([
+                new ReferenceConfigurator('sonata.admin.pool'),
+                new ReferenceConfigurator('sonata.admin.breadcrumbs_builder'),
+            ])
+
+        // NEXT_MAJOR: Remove this service.
+        ->set('sonata.admin.controller.admin', HelperController::class)
+            ->public()
+            ->deprecate('The controller service "%service_id%" is deprecated in favor of several action services since version 3.38.0 and will be removed in 4.0.')
+            ->args([
+                new ReferenceConfigurator('twig'),
+                new ReferenceConfigurator('sonata.admin.pool'),
+                new ReferenceConfigurator('sonata.admin.helper'),
+                new ReferenceConfigurator('validator'),
+            ])
+
+        ->set('sonata.admin.audit.manager', AuditManager::class)
+            ->public()
+            ->args([
+                new ReferenceConfigurator('service_container'),
+            ])
+
+        ->alias(AuditManager::class, 'sonata.admin.audit.manager')
+
+        ->alias(AuditManagerInterface::class, 'sonata.admin.audit.manager')
+
+        // NEXT_MAJOR: Remove this service.
+        ->set('sonata.admin.exporter', Exporter::class)
+            ->public()
+            ->deprecate('The service "%service_id%" is deprecated in favor of the "sonata.exporter.exporter" service since version 3.14 and will be removed in 4.0.')
```

Comment:
```suggestion
            ->deprecate('The service "%service_id%" is deprecated since sonata-project/admin-bundle 3.14.0 and will be removed in 4.0. Use "sonata.exporter.exporter" service instead.')
```